modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ObjectEntryDisplayContext.java

```diff
@@ -329,21 +333,76 @@ public ObjectLayoutTab getObjectLayoutTab() throws PortalException {
 		return optional.get();
 	}
 
-	public CreationMenu getRelatedModelCreationMenu() throws PortalException {
+	public CreationMenu getRelatedModelCreationMenu(
+			ObjectDefinition objectDefinition2)
+		throws PortalException {
+
 		if (_readOnly || isDefaultUser()) {
 			return null;
 		}
 
+		ObjectDefinition objectDefinition1 = getObjectDefinition();
+
 		LiferayPortletResponse liferayPortletResponse =
 			_objectRequestHelper.getLiferayPortletResponse();
 
+		if (objectDefinition1.isSystem() || objectDefinition2.isSystem() ||
+			(StringUtil.equals(
+				objectDefinition1.getScope(),
+				ObjectDefinitionConstants.SCOPE_COMPANY) &&
+			 StringUtil.equals(
+				 objectDefinition2.getScope(),
+				 ObjectDefinitionConstants.SCOPE_SITE))) {
+
+			return CreationMenuBuilder.addDropdownItem(
+				dropdownItem -> {
+					dropdownItem.setHref(
+						liferayPortletResponse.getNamespace() +
+							"selectRelatedModel");
+					dropdownItem.setLabel(
+						LanguageUtil.get(
+							_objectRequestHelper.getRequest(),
+							"select-existing-one"));
+					dropdownItem.setTarget("event");
+				}
+			).build();
+		}
+
+		ServiceContext serviceContext =
+			ServiceContextThreadLocal.getServiceContext();
+
 		return CreationMenuBuilder.addDropdownItem(
+			dropdownItem -> {
+				dropdownItem.setHref(
+					PortletURLBuilder.create(
+						PortalUtil.getControlPanelPortletURL(
+							_objectRequestHelper.getRequest(),
+							serviceContext.getScopeGroup(),
+							objectDefinition2.getPortletId(), 0, 0,
+							PortletRequest.RENDER_PHASE)
+					).setMVCRenderCommandName(
+						"/object_entries/edit_object_entry"
+					).setBackURL(
+						_objectRequestHelper.getCurrentURL()
+					).setParameter(
+						"objectDefinitionId",
+						objectDefinition2.getObjectDefinitionId()
+					).setWindowState(
+						WindowState.MAXIMIZED
+					).buildString());
+				dropdownItem.setLabel(
+					LanguageUtil.get(
+						_objectRequestHelper.getRequest(), "create-new"));
```

Comment:
Hey @seltong , I think you should validate if the user has the permission do add entry before adding the option to menu.
We have this acceptance criteria in the story: "The create new option should only appear if the user has permission to add entries in the related object"